public/app/plugins/datasource/prometheus/dataquery.gen.ts

```diff
@@ -0,0 +1,36 @@
+// Code generated - EDITING IS FUTILE. DO NOT EDIT.
+//
+// Generated by:
+//     public/app/plugins/gen.go
+// Using jennies:
+//     TSTypesJenny
+//     PluginTSTypesJenny
+//
+// Run 'make gen-cue' from repository root to regenerate.
+
+import * as common from '@grafana/schema';
+
+export const DataQueryModelVersion = Object.freeze([0, 0]);
+
+export enum QueryEditorMode {
+  Builder = 'builder',
+  Code = 'code',
+}
+
+export interface Prometheus extends common.DataQuery {
+  /**
+   * Code or Builder mode
+   */
+  editorMode?: QueryEditorMode;
+  exemplar?: boolean;
+  expr: string;
+  format?: string;
+  hinting?: boolean;
+  instant?: boolean;
+  interval?: string;
+  intervalFactor?: number;
+  intervalMs?: number;
+  range?: boolean;
+  showingGraph?: boolean;
+  showingTable?: boolean;
```

Comment:
We should check and decide which one should be optional and which one should be required.

Reply:
Yes, having a discussion about what is optional and required would be helpful. As I have worked on migrations to React in other datasources, a number of fields have been added to the dataquery model to handle the migration, and all of these are optional. The most important field is `expr` and everything else is feature based, but it would help to go through all of these one by one and confirm that.

Reply:
range, interval, intervalMs, and expr